src/core/components/InstallButton/index.js

```diff
@@ -72,6 +84,32 @@ export class InstallButtonBase extends React.Component {
           {i18n.gettext('Install Theme')}
         </Button>
       );
+    } else if (addon.type === ADDON_TYPE_OPENSEARCH) {
+      const onClick = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        if (
+          !buttonIsDisabled && _window.external &&
+          'AddSearchProvider' in _window.external
+        ) {
+          _log.info('Adding OpenSearch Provider', { addon });
+          _window.external.AddSearchProvider(addon.installURL);
```

Comment:
is there any way to pass a callback or something in case of an error? At the very least we should log errors. But I don't see any way to do it if these docs are accurate: https://developer.mozilla.org/en-US/docs/Web/API/Window/sidebar/Adding_search_engines_from_Web_pages